src/testplane.ts

```diff
@@ -101,7 +112,29 @@ export class Testplane extends BaseTestplane {
         );
         this.runner = runner;
 
-        this.on(MasterEvents.TEST_FAIL, () => this._fail()).on(MasterEvents.ERROR, (err: Error) => this.halt(err));
+        this.on(MasterEvents.TEST_FAIL, () => this._fail());
+        this.on(MasterEvents.TEST_FAIL, res => this._addFailedTest(res));
+        this.on(MasterEvents.ERROR, (err: Error) => this.halt(err));
+
+        if (runFailed) {
+            let previousRunFailedTests: FailedListItem[];
+            this.on(MasterEvents.INIT, async () => {
+                previousRunFailedTests = await this._readFailed();
+            });
+
+            this.on(MasterEvents.AFTER_TESTS_READ, testCollection => {
+                if (previousRunFailedTests.length) {
+                    testCollection.disableAll();
+                    previousRunFailedTests.forEach(({ fullTitle, browserId }) => {
+                        testCollection.enableTest(fullTitle, browserId);
+                    });
+                }
```

Comment:
Also, if run-failed option was passed, but no failed tests file was found, should we print a warning about it, what do you think?